library/oneview_logical_interconnect.py

```diff
@@ -300,42 +300,39 @@ class LogicalInterconnectModule(OneViewModuleBase):
     def __init__(self):
         super(LogicalInterconnectModule, self).__init__(additional_arg_spec=self.argument_spec,
                                                         validate_etag_support=True)
-        self.resource_client = self.oneview_client.logical_interconnects
+        self.set_resource_object(self.oneview_client.logical_interconnects)
 
     def execute_module(self):
 
-        resource = self.__get_by_name(self.data)
-
-        if not resource:
+        if not self.current_resource:
             raise OneViewModuleResourceNotFound(self.MSG_NOT_FOUND)
 
         changed, msg, ansible_facts = False, '', dict()
-        uri = resource['uri']
 
         if self.state == 'compliant':
-            changed, msg, ansible_facts = self.__compliance(uri)
+            changed, msg, ansible_facts = self.__compliance()
```

Comment:
Can we rename this also to `update_compliance`?

Reply:
This method is call to manage the state, its matching with the state name.